.travis.yml

```diff
@@ -33,21 +32,26 @@ env:
     - BUCKET=PPS5
     - BUCKET=PPS6
     - BUCKET=EXAMPLES
+
+stages:
+  - build
+  - test
+
+jobs:
+  include:
+    - stage: build
+      script: etc/testing/travis_build.sh
+
 before_install:
   # Make sure cache dirs exist and are writable
   - etc/testing/travis_cache.sh
-  - etc/testing/travis_before_install.sh
 install:
   - etc/testing/travis_install.sh
-before_script:
-  - docker version
-script:
-  - etc/testing/travis.sh
+script: etc/testing/travis.sh
 before_cache:
```

Comment:
Same thing here. Do we need before_cache step?

Reply:
Yes, we need both because they both need to run `travis_cache.sh`. While we're no longer caching the docker build container, other directories are still being cached - and due to a TravisCI bug, we need to fix permissions on them for caching to work.